gui/src/components/onboarding/pages/trackers-assign/TrackerAssignment.tsx

```diff
@@ -46,27 +48,54 @@ interface FlatDeviceTrackerDummy {
   };
 }
 
+// Ordered collection of assign modes with the number of IMU trackers
+const ASSIGN_MODE_OPTIONS = [
+  AssignMode.LowerBody,
+  AssignMode.Core,
+  AssignMode.EnhancedCore,
+  AssignMode.FullBody,
+  AssignMode.All,
+].reduce(
+  (options, mode) => ({ ...options, [mode]: ASSIGNMENT_MODES[mode].length }),
+  {} as Record<AssignMode, number>
+);
+
 export function TrackersAssignPage() {
   const { isMobile } = useBreakpoint('mobile');
   const { l10n } = useLocalization();
   const { config, setConfig } = useConfig();
-  const { useAssignedTrackers, trackers } = useTrackers();
+  const { useAssignedTrackers, useConnectedIMUTrackers, trackers } =
+    useTrackers();
   const { applyProgress, state } = useOnboarding();
   const { sendRPCPacket, useRPCPacket } = useWebsocketAPI();
   const defaultValues = {
-    advanced: config?.advancedAssign ?? false,
+    assignMode: config?.assignMode ?? AssignMode.Core,
     mirrorView: config?.mirrorView ?? true,
   };
-  const { control, watch } = useForm<{
-    advanced: boolean;
+  const { control, watch, setValue } = useForm<{
+    assignMode: AssignMode;
     mirrorView: boolean;
   }>({ defaultValues });
-  const { advanced, mirrorView } = watch();
+  const { assignMode, mirrorView } = watch();
   const [selectedRole, setSelectRole] = useState<BodyPart>(BodyPart.NONE);
   const assignedTrackers = useAssignedTrackers();
   useEffect(() => {
-    setConfig({ advancedAssign: advanced, mirrorView });
-  }, [advanced, mirrorView]);
+    setConfig({ assignMode, mirrorView });
+  }, [assignMode, mirrorView]);
+
+  const connectedIMUTrackers = useConnectedIMUTrackers().length;
+  useEffect(() => {
+    if (connectedIMUTrackers <= ASSIGN_MODE_OPTIONS[assignMode]) return;
+
+    const selectedAssignMode =
+      (Object.entries(ASSIGN_MODE_OPTIONS).find(
+        ([_, count]) => count >= connectedIMUTrackers
+      )?.[0] as AssignMode) ?? AssignMode.All;
+
+    if (assignMode !== selectedAssignMode) {
+      setValue('assignMode', selectedAssignMode);
+    }
+  }, [connectedIMUTrackers]);
```

Comment:
shouldn't the effect have `assignMode` as a dependency if it checks it?
```suggestion
  }, [connectedIMUTrackers, assignMode]);
```